src/test/java/api/support/matchers/JsonObjectMatcher.java

```diff
@@ -0,0 +1,57 @@
+package api.support.matchers;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.hamcrest.Description;
+import org.hamcrest.Matcher;
+import org.hamcrest.TypeSafeDiagnosingMatcher;
+
+import com.jayway.jsonpath.ReadContext;
+import com.jayway.jsonpath.matchers.JsonPathMatchers;
+
+import io.vertx.core.json.JsonObject;
+
+public class JsonObjectMatcher extends TypeSafeDiagnosingMatcher<JsonObject> {
+
+  @SafeVarargs
+  public static Matcher<JsonObject> allOfPaths(Matcher<? super ReadContext>... jsonPathMatchers) {
```

Comment:
What is the `ReadContext`?

Reply:
It's type parameter for the matchers  from `json-path-assert` library. Here is a signature: 
`Matcher<? super ReadContext> withJsonPath(String jsonPath, Matcher<T> resultMatcher)`
It is usually used as:
`String json = "{\"empty_object\": {}}";`
`assertThat(json, isJsonString(withJsonPath("$.empty_object")));`
So that `isJsonString` maps `Matcher<? super ReadContext>` to `Matcher<String>`.
And `JsonObjectMatcher#allOfPaths` (implemented in this pull request) applies `isJsonString` to all the matchers, and matches actual value  against them.

Reply:
The option was to take `Matcher<String>... jsonPathMatchers` as a parameter, but it would require to wrap all the 'withJsonPath()' with 'isJsonString()'